include/bouncer.h

```diff
@@ -516,6 +517,12 @@ struct PgCredentials {
 	 * settings and connection count tracking.
 	 */
 	PgGlobalUser *global_user;
+
+	int iterations;
+	char *salt;	/* base64-encoded */
+	uint8_t StoredKey[SCRAM_KEY_LEN];
+	uint8_t ServerKey[SCRAM_KEY_LEN];
```

Comment:
How is this different from the `scram_ServerKey` field in this struct? Do we need both. If so We need some comments about how they are different.

Reply:
I've changed to reuse the scram_ServerKey and scram_ClientKey and changed the has_scram_keys to use_scram_keys for the scram pass-through feature. The adhoc_scram_secrets_cached controls if scram_ServerKey,  scram_StoredKey and scram_SaltKey should be re-used when building the scram message exchange.